addons/binding/org.openhab.binding.netatmo/src/main/java/org/openhab/binding/netatmo/internal/thermostat/NATherm1Handler.java

```diff
@@ -135,31 +231,51 @@ public void handleCommand(ChannelUID channelUID, Command command) {
-                                null, null);
-
-                        updateState(channelUID, new StringType(command.toString()));
-                        requestParentRefresh();
+                        String target_mode = command.toString();
+                        if (CHANNEL_SETPOINT_MODE_MANUAL.equals(target_mode)) {
+                            logger.info(
+                                    "Switching to manual mode is done by assigning a setpoint temperature - command dropped");
+                            updateState(channelUID, getSetpoint());
+                        } else {
+                            pushSetpointUpdate(target_mode, null, null);
+                        }
                         break;
                     }
                     case CHANNEL_SETPOINT_TEMP: {
-                        // Switch the thermostat to manual mode on the desired setpoint for given duration
-                        Calendar cal = Calendar.getInstance();
-                        cal.add(Calendar.MINUTE, getSetPointDefaultDuration());
-                        getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), "manual",
-                                (int) (cal.getTimeInMillis() / 1000), Float.parseFloat(command.toString()));
-                        updateState(channelUID, new DecimalType(command.toString()));
-                        requestParentRefresh();
+                        pushSetpointUpdate(CHANNEL_SETPOINT_MODE_MANUAL, getSetpointEndTime(),
+                                Float.parseFloat(command.toString()));
                         break;
                     }
+                    case CHANNEL_PLANNING: {
+                        getBridgeHandler().getThermostatApi().switchschedule(getParentId(), getId(),
+                                command.toString());
+                        updateState(channelUID, new StringType(command.toString()));
+                        invalidateParentCacheAndRefresh();
+                    }
                 }
             } catch (Exception e) {
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.NONE, e.getMessage());
             }
         }
     }
 
+    private void pushSetpointUpdate(String target_mode, Integer setpointEndtime, Float setpointTemp) {
+        getBridgeHandler().getThermostatApi().setthermpoint(getParentId(), getId(), target_mode, setpointEndtime,
+                setpointTemp);
+
+        invalidateParentCacheAndRefresh();
+    }
+
+    private int getSetpointEndTime() {
+        Calendar cal = Calendar.getInstance();
+        cal.add(Calendar.MINUTE, getSetPointDefaultDuration());
+        cal.set(Calendar.SECOND, 0);
+        cal.set(Calendar.MILLISECOND, 0);
+        return (int) (cal.getTimeInMillis() / 1000);
+    }
+
     private int getSetPointDefaultDuration() {
+        // TODO : this informations could be sourced from Netatmo API instead of a local configuration element
```

Comment:
There is a TODO, what should be done and can it be completed before merging the PR?

Reply:
The swagger definition file has to be heavily updated. Netatmo API has grantly evolved these past month due to the introduction of valves, presence... I plan to revamp the swagger definition but it will take some time. So this todo is a reminder for the future.

Reply:
👍